### PR TITLE
precourse exercices

### PR DESCRIPTION
**Pull request by adria-zhu**
You have the update version of the code named as ¨update code¨ in down below

### CODE REVIEW

File: staff/adrian-zhu/precourse/Airflight/index.js [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        }
+        
+        showOptionsAdmin();
+    }else if(option=='Eliminar'|| option =='Eliminar'|| option=='eliminar'){
+        let flightId=prompt('Por favor diganós el id del vuelo que usted quieres eliminar');
+        deleteFlight(flightId);
+        showFlights(flights);
+        showOptionsAdmin();
+    }else if(option==null){
+        alert('Hasta luego!')
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Airflight/index.js]: Missing semicolon

File: staff/adrian-zhu/precourse/Airflight/index.js [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        
+        if(flights[i].id==idFl){
+            alert(`Has eliminado a ${flights[i].id}`);
+            flights.splice(i, 1);
+            flag=false;
+        }
+
+    }
+    if (flag==true){
+        alert(`Lo sentimos, no hemos podido eliminar a este ${idFl} que usted has indicado. `)
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Airflight/index.js]: Missing semicolon

File: staff/adrian-zhu/precourse/Airflight/index.js [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    }
+} else{
+    alert('Los vuelos están eliminado correctamente.')
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Airflight/index.js]: semicolon

File: staff/adrian-zhu/precourse/Calculadora/index.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+function calculator () {
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Calculadora/index.js:1]: missing semiclons in entire file

File: staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/index.html [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    <meta charset="UTF-8">
+<meta name="viewport" content="width=device-width, initial-scale=1.0">
+<meta http-equiv="X-UA-Compatible" content="ie=edge">
+<title>Pasapalabra</title>
+<link rel="stylesheet" href="./css/style.css">
+</head>
+<body>  
+	<section id = ¨part-1¨>
+        <div class="alert">
+        <span class="closebtn" onclick="this.parentElement.style.display='none';">&times;</span>
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/index.html]: Should be and eventListener in .js

File: staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+// DOM: capturar informaciones
+var input = document.getElementById("respuesta");
+var definicion =document.getElementById("definicion");
+var correcta = document.getElementById("correcta"); 
+
+// DOM: btn
+var btnInput = document.getElementById("btnrespuesta");
+var btnAbandonar = document.getElementById("btnAbandonar"); // btn-abandonar
+var btnInformación = document.getElementById("btnInformación");
+var btnJugador = document.getElementById("btnJugador")
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js]: missing semicolon

File: staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+            i++;
+        }
+        else{break;}
+    }
+}
+
+function logica(){
+    if(i<27){
+        definicion.innerHTML = questions[i].question;
+        document.getElementById(questions[i].letter).style.backgroundColor = '#64b5f6'
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js]: Missing semicolon

File: staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+        document.getElementById(questions[i].letter).style.border = "2px solid black";
+        }
+    else{
+        clearTimeout(myVar);
+        clearTimeout(myVar2);
+        final()};
+}
+
+function colorAzul(){
+    document.getElementById(questions[i].letter).style.backgroundColor = '#1976d2'
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js]: Missing semicolon

File: staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+    document.getElementById(questions[i].letter).style.transition = 'background-color 1s, border 1s, color 1s';
+}
+
+function checkReplay(respuesta,indice){
+    
+    if(respuesta.value.toLowerCase()==="pasapalabra"){
+        colorAzul()
+        correcta.innerHTML= "Has clicado a Pasapalabra"
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js]: Missing semicolon

File: staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+}
+
+
+function pasapalabra(){
+    colorAzul();
+    buscadorPregunas();
+    i++;
+    input.value ="";
+    logica();
+    correcta.innerHTML= "Has clicado a Pasapalabra"
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js]: Missing semicolon

File: staff/adrian-zhu/precourse/Airflight/index.js [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+
+    for(let i=0;i<lastFlights.length; i++){
+        console.log(`To ${lastFlights[i].to}`);
+    }
+    return
+}
+function askProfileUser(){
+
+    let profile=prompt('Es su perfil administrador o usuario?');
+    if(profile=='ADMINISTRADOR'|| profile=='administrador' || profile=='Administrador'){
```
Comment (ElenaErodrial) [inline staff/adrian-zhu/precourse/Airflight/index.js]: Could have done .toLowerCase() and avoid having to write all the different options (||).

File: staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+    btnPasapalabra.disabled = true;
+    btnJugador.disabled = false;
+    
+    correcta.innerHTML=`Has terminado el juego tienes ${correct} preguntas correctas y ${incorrect} preguntas incorrectas`
+    btnInformación.innerHTML=`Dale otra partida`;
+}
+
+function init(){
+
+    estadoInicio()
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js]: Missing semicolon

File: staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+    btnInput.disabled = false;
+    btnAbandonar.disabled = false;
+    btnPasapalabra.disabled = false;
+
+    btnJugador.disabled = true;
+    
+    // html ¨a¨ preset
+    definicion.innerHTML = questions[i].question;
+    document.getElementById("a").style.border = "2px solid floralwhite";
+    document.getElementById("a").style.backgroundColor = '#64b5f6'
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Pasapalabra HTML+CSS/js/pasapalabra2.js]: Missing semicolon

File: staff/adrian-zhu/precourse/Pasapalabra/index.js [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+        { letter: "y", respuesta: "peyote", status: 0, pregunta: ("Contiene la Y. Pequeño cáctus conocido por sus alcaloides psicoactivos utilizado de forma ritual y medicinal por indígenas americanos") },
+        { letter: "z", respuesta: "zen", status: 0, pregunta: ("Con la Z. Escuela de budismo que busca la experiencia de la sabiduría más allá del discurso racional") }
+      ],
+      correcto: [],
+      incorrecto: []
+    }
+  
+    function play() {
+      var posicion = 0;
+      let intentos = 1
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Pasapalabra/index.js]: Missing semicolon

File: staff/adrian-zhu/precourse/Pasapalabra/index.js [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+      alert(`Bienvenidos check Skylab-pasapalabra`)
+
+      function check() {
+        var pregunta = darpregunta(posicion);
+        var respuesta = darrespuesta(posicion);
+
+        console.log(pregunta)
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Pasapalabra/index.js]: Missing semicolon

File: staff/adrian-zhu/precourse/Pasapalabra/index.js [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+
+       // console.log(`Te queda ${preguntas.pendiente.length} preguntas`);
+        
+        var letra = prompt("escribe-nos una de las opciones:\nRespuesta de la pregunta/Pasapalabra/Exit");
+        var status = checkrespuesta(letra.toLowerCase(), respuesta, posicion);
+        console.log(status)
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Pasapalabra/index.js]: Missing semicolon

File: staff/adrian-zhu/precourse/Pasapalabra/index.js [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+      console.log("Respuesta correcta: " + respuesta);
+      console.log("Tu respuesta: " + letra);
+  
+      if(letra == "exit") {
+        status = 0;
+      } else if(letra == "pasapalabra") {
+        preguntas.pendiente[posicion].status = 2
+      } else if(letra == respuesta) {
+        preguntas.correcto.push(preguntas.pendiente[posicion]);
+        //preguntas.pendiente.splice(posicion, 1);
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Pasapalabra/index.js]: Useless comment should not be in deploy code

File: staff/adrian-zhu/precourse/Airflight/index.js [hunk 18]
```diff
@@ -0,0 +1,231 @@
...
+    console.log(`<--------------------------------------------------------------->`);
+    lastFlights();
+    let prof=askProfileUser();
+}
+
+function withStop(){
+    
+    for(let i=0;i<flights.length; i++){
+
+        if(flights[i].scale==true){
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Airflight/index.js:22]: In conditional It's the same   if(flights[i].scale==true)   than   if(flights[i].scale)

File: staff/adrian-zhu/precourse/Airflight/index.js [hunk 19]
```diff
@@ -0,0 +1,231 @@
...
+        askProfileUser();
+    }
+    return
+}
+
+function showOptionsAdmin(){
+
+    let option=prompt('Encantad@ administrador, usted puedes insertar o eliminar un vuelo');
+
+    if (option=='INSERTAR'|| option=='Insertar'|| option=='insertar'){
```
Comment (martin-garcia-blanco) [inline staff/adrian-zhu/precourse/Airflight/index.js]: Take care with comparations, Its better to parse to upperCase and compare then

File: staff/adrian-zhu/precourse/Airflight/index.js (hunk 18 above)
Comment (adria-zhu) [reply staff/adrian-zhu/precourse/Airflight/index.js:22]: ok

File: staff/adrian-zhu/precourse/Airflight/index.js (hunk 11 above)
Comment (adria-zhu) [reply staff/adrian-zhu/precourse/Airflight/index.js]: Thanks, it is fixed

File: staff/adrian-zhu/precourse/Airflight/index.js [hunk 20]
```diff
@@ -0,0 +1,232 @@
...
+
+}
+function lastFlights(){
+
+    let lastFlights=flights.slice(5,10);
+
+    for(let i=0;i<lastFlights.length; i++){
+        console.log(`To ${lastFlights[i].to}`);
+    }
+    return
```
Comment (ghost) [inline staff/adrian-zhu/precourse/Airflight/index.js:73]: This function only runs correctly if you have 10 flights in the flights array, maybe useful specificly for the PRO version

**Pull request merged**